tests/cross/src/test/scala/tests/pc/CompletionArgSuite.scala

```diff
@@ -699,4 +707,252 @@ class CompletionArgSuite extends BaseCompletionSuite {
+  )
+
+  check(
+    "overloaded-methods",
+    """|class A() {
+       |  def m(anInt : Int): Int = ???
+       |  def m(aString : String): String = ???
+       |}
+       |object O {
+       |  def m(aaa: Int): Int = ???
+       |  val k = new A().m(a@@)
+       |}
+       |""".stripMargin,
+    """|aString = : String
+       |anInt = : Int
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  // type of `myInstance` is resolved to `null` for Scala 2
+  check(
+    "overloaded-methods2".tag(IgnoreScala2),
+    """|class A() {
+       |  def m(anInt : Int): Int = ???
+       |  def m(aString : String): String = ???
+       |  private def m(aBoolean: Boolean): Boolean = ???
+       |}
+       |object O {
+       |  def m(aaa: Int): Int = ???
+       |  val myInstance = new A()
+       |  val k = myInstance.m(a@@)
+       |}
+       |""".stripMargin,
+    """|aString = : String
+       |anInt = : Int
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  check(
+    "overloaded-select",
+    """|package a.b {
+       |  object A {
+       |    def m(anInt : Int): Int = ???
+       |    def m(aString : String): String = ???
+       |  }
+       |}
+       |object O {
+       |  def m(aaa: Int): Int = ???
+       |  val k = a.b.A.m(a@@)
+       |}
+       |""".stripMargin,
+    """|aString = : String
+       |anInt = : Int
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  check(
+    "overloaded-in-a-class",
+    """|trait Planet
+       |case class Venus() extends Planet
+       |class Main[T <: Planet](t : T) {
+       |  def m(inn: Planet, abb: Option[Int]): Int = ???
+       |  def m(inn: Planet, aaa: Int): Int = ???
+       |  def k = m(t, a@@)
+       |}
+       |""".stripMargin,
+    """|aaa = : Int
+       |abb = : Option[Int]
+       |""".stripMargin,
+    topLines = Some(2),
+  )
+
+  check(
+    "overloaded-function-param".tag(IgnoreScala2),
+    """|def m[T](i: Int)(inn: T => Int, abb: Option[Int]): Int = ???
+       |def m[T](i: Int)(inn: T => Int, aaa: Int): Int = ???
+       |def m[T](i: Int)(inn: T => String, acc: List[Int]): Int = ???
+       |def k = m(1)(inn = identity[Int], a@@)
+       |""".stripMargin,
+    """|aaa = : Int
+       |abb = : Option[Int]
+       |assert(assertion: Boolean): Unit
+       |""".stripMargin,
+    topLines = Some(3),
+  )
+
+  check(
+    "overloaded-function-param2".tag(IgnoreScala2),
+    """|def m[T](i: Int)(inn: T => Int, abb: Option[Int]): Int = ???
+       |def m[T](i: Int)(inn: T => Int, aaa: Int): Int = ???
+       |def m[T](i: String)(inn: T => Int, acc: List[Int]): Int = ???
+       |def k = m(1)(inn = identity[Int], a@@)
+       |""".stripMargin,
+    """|aaa = : Int
+       |abb = : Option[Int]
+       |assert(assertion: Boolean): Unit
+       |""".stripMargin,
+    topLines = Some(3),
+  )
+
+  // doesn't filter properly for Scala 2, since the filtering heuristic for matching methods
+  // isn't accurate enough (polymorphic args types are resolved to `null`)
```

Comment:
Also look suspect, but maybe this needs a workaround? You could gather up the remaining cases in an issue, really for any ignored Scala 2 cases that do not contain any Scala 3 specific code.